src/runtime/tests/manifest-parser-test.ts

```diff
@@ -284,6 +284,30 @@ describe('manifest parser', () => {
         outRef: writes Reference<Bar>
     `);
   });
+  it('parses refinement types in a schema', () => {
+      parse(`
+        schema Foo
+          num: Number [num > 10]
+      `);
+  });
+  it('parses refinement types in a particle', () => {
+    parse(`
+    particle Foo
+      input: reads Something {value: Text [value is lower case]}
+    `);
+  });
+  it('does not parse sq brackets in the refinement expression', () => {
+    //TODO(ragdev): Improve the error message to be more accurate as the expression syntax evolves.
+      let data;
+      assert.throws(() => {
+      const manifestAst = parse(`
+      particle Foo
+        input: reads Something {value: Text [value [] is lower case]}
+      `);
+      data = JSON.parse(manifestAst[1].data);
+      }, `Expected "alias", "external", "import", "meta", "particle", "recipe", "resource", "schema", "store", a particle item, a trigger for a recipe, an annotation (e.g. @foo), or an interface but "i" found.`);
+      assert.strictEqual(data, undefined);
+  });
```

Comment:
I'd leave this out until the error message is the one you expect.

Reply:
I'd prefer that we kept it so that if we accidentally break the refinement expression parsing this catches us.

It would be very easy to have the parsing consume too much text while we iterate on the grammar.

Reply:
👍